mite/logoutput.py

```diff
@@ -173,7 +192,66 @@ def min_resp_time(self):
     @property
     def req_sec_mean(self):
         return self._req_total / (self._scenarios_completed_time - self._init_time)
-
+    
+    # @property
+    # def resp_time_standard_deviation(self):
+    #     if not self._resp_time_store:
+    #         return 0
+    #     variance = []
+    #     for resp_time in self._resp_time_store:
+    #         variance.append(math.sqrt((resp_time-self.mean_resp_time)**2))
+    #     std_deviation = sum(variance) / len(self._resp_time_store)
+    #     print(f"resp time is {self._resp_time_store}")
+    #     return std_deviation
+    
+    @property
+    def resp_time_standard_deviation(self):
+        if not self._resp_time_store:
+            return 0
+        return np.std(self.resp_time_array)
+    
+    @property
+    def req_sec_standard_deviation(self):
+        if self._req_total == 0:
+            return 0
+        variance = []
+        for req_sec in self._req_sec_store:
+            variance.append(math.sqrt((req_sec-self.req_sec_mean)**2))
+        std_deviation = sum(variance) / len(self._req_sec_store)
+        return std_deviation
+    
+    # @property
+    # def resp_time_within_standard_deviation(self):
+    #     upper_limit = self.mean_resp_time + self.resp_time_standard_deviation
+    #     # lower_limit = self.mean_resp_time - self.resp_time_standard_deviation
+    #     count = 0
+    #     for resp_time in self._resp_time_store:
+    #         if (upper_limit >= resp_time): # and (resp_time >= lower_limit):
+    #             count = count +1
+    #     stats_percentage = count / len(self._resp_time_store) * 100
+    #     # print(f"\nresp_time_store is {self._resp_time_store} \n")
+    #     # print(f"len of resp time store is {len(self._resp_time_store)}\n")
+    #     # del self._resp_time_store[:]
+    #     return stats_percentage
+    
+    @property
+    def resp_time_within_standard_deviation(self):
+        if not self._resp_time_store:
+            return 0
+        return np.mean(((self.resp_time_array <= (self.mean_resp_time + self.resp_time_standard_deviation)) & (self.resp_time_array >= (self.mean_resp_time - self.resp_time_standard_deviation)))* 100)
```

Comment:
i have no idea what this algorithm is doing...can you explain what it's trying to calculate?  you seem to be mixing booleans with bitwise operators, which points to very subtle things going on....

Reply:
It is going to return the percentage of values that lie between the upper and lower limits of deviation. 

(self.resp_time_array <= (self.mean_resp_time + self.resp_time_standard_deviation)) & (self.resp_time_array >= (self.mean_resp_time - self.resp_time_standard_deviation) creates a boolean array where each element is True if the condition is met and False otherwise. np.mean should return the fraction of True values

Reply:
I see.  Could we use something like `abs(x - self.mean_resp_time) < self.std_dev`?  It may be a matter of taste (so feel free to tell me you disagree), but at least to me that's a clearer way of writing the algorithm than `mean-stddev < x < mean+stddev`.

Whatever way we decide to code it, I think a comment explaining what the calculation does would be really helpful too.

Reply:
This is actually good and can be easily understood. I'll make changes